instrumentation/github.com/99designs/gqlgen/otelgqlgen/config.go

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelgqlgen
+
+import (
+	"go.opentelemetry.io/otel/trace"
+)
+
+// config is used to configure the mongo tracer.
+type config struct {
+	TracerProvider          trace.TracerProvider
+	Tracer                  trace.Tracer
+	ComplexityExtensionName string
+}
+
+// Option specifies instrumentation configuration options.
+type Option func(*config)
```

Comment:
The guidelines around the `Option` type have changed a little.

Please see: https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#option

Here is a sample PR how you can easily make it compliant: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/860